go/libraries/doltcore/sqle/writer/prolly_write_session.go

```diff
@@ -193,5 +193,25 @@ func (s *prollyWriteSession) setWorkingSet(ctx context.Context, ws *doltdb.Worki
 		delete(s.tables, name)
 	}
 	s.workingSet = ws
+
+	root := ws.WorkingRoot()
+	if err := s.updateAutoIncrementSequences(ctx, root); err != nil {
+		return err
+	}
+
 	return nil
 }
+
+func (s *prollyWriteSession) updateAutoIncrementSequences(ctx context.Context, root *doltdb.RootValue) error {
```

Comment:
I don't think this is the behavior we want for auto increment. In particular, rolling back a transaction does not reset the auto increment sequence to its previous state

Reply:
@zachmu